src/pydvl/utils/parallel/map_reduce.py

```diff
@@ -175,24 +158,21 @@ def __init__(
 
     def __call__(
         self,
-        inputs: Union[Collection[T], Any],
-        *,
-        n_jobs: Optional[int] = None,
-        n_runs: Optional[int] = None,
-        chunkify_inputs: Optional[bool] = None,
+        inputs: Union[Sequence[T], Sequence[Utility], Utility],
```

Comment:
Is there a reason not to pass the inputs to the constructor instead? IMO, a "Job" should contain all information required to run it, and any `call` or `run` method wouldn't take any arguments.

Also, see my other comment about accepting non-sequence objects. At first sight, it doesn't make much sense to accept single items for a parallel job and it doesn't take any effort for the user of the class to wrap one in a tuple in case it's needed